rust/sdk-processor/src/processors/ans_processor.rs

```diff
@@ -0,0 +1,158 @@
+use crate::{
+    config::{
+        db_config::DbConfig,
+        indexer_processor_config::IndexerProcessorConfig,
+        processor_config::{DefaultProcessorConfig, ProcessorConfig},
+    },
+    steps::{
+        ans_processor::{AnsExtractor, AnsStorer},
+        common::get_processor_status_saver,
+    },
+    utils::{
+        chain_id::check_or_update_chain_id,
+        database::{new_db_pool, run_migrations, ArcDbPool},
+        starting_version::get_starting_version,
+    },
+};
+use ahash::AHashMap;
+use anyhow::Result;
+use aptos_indexer_processor_sdk::{
+    aptos_indexer_transaction_stream::{TransactionStream, TransactionStreamConfig},
+    builder::ProcessorBuilder,
+    common_steps::{
+        TransactionStreamStep, VersionTrackerStep, DEFAULT_UPDATE_PROCESSOR_STATUS_SECS,
+    },
+    traits::{processor_trait::ProcessorTrait, IntoRunnableStep},
+};
+use processor::worker::TableFlags;
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use tracing::{debug, info};
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct AnsProcessorConfig {
+    // #[serde(flatten)]
+    // pub default: DefaultProcessorConfig,
```

Comment:
Is there a better way to do this? The serde flatten wasn't working out for whatever reason @rtso @yuunlimm